nixos/doc/manual/development/replace-modules.xml

```diff
@@ -0,0 +1,48 @@
+<section xmlns="http://docbook.org/ns/docbook"
+        xmlns:xlink="http://www.w3.org/1999/xlink"
+        xmlns:xi="http://www.w3.org/2001/XInclude"
+        version="5.0"
+        xml:id="sec-replace-modules">
+
+<title>Replace Modules</title>
+
+<para>It's also possible to disable modules, removing the option
+  declarations and config implementation defined within that file.
+  This can be used to import a set of modules from another channel
+  while keeping the rest of the system on a stable release.</para>
+<para><literal>disabledModules</literal> is a top level attribute like
+  <literal>options</literal> and <literal>config</literal>.  It
+  contains a list of modules that will be disabled. This can either be
+  the full path to the module or a string with the filename relative
+  to the modules path (eg. &lt;nixpkgs/nixos/modules&gt; for nixos).
+  </para>
+
+<programlisting>
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+
+  cfg = config.services.postgresql;
+
+in
+
+{
+  disabledModules = [ "services/databases/postgresql.nix" ];
```

Comment:
nit: Disabling a module implies that you at least declare the equivalent set of option, otherwise you will probably have error messages because options are defined but never declared.  I suggest you take a smaller existing module than `postgresql`.